lexer.c

```diff
@@ -235,7 +209,7 @@ static Token *read_token_int(void)
 
 bool is_punct(Token *tok, int c)
 {
-    return tok && (tok->type == TTYPE_PUNCT) && (tok->punct == c);
+    return tok && (tok->type == TTYPE_PUNCT) && ((int) tok->priv == c);
```

Comment:
How about providing some helper functions/macros to map the  operations such as `(char) tok->priv)`? Thus, we don't have to manipuate with `priv`.

Reply:
Yes, a helper function for private data had planned, and the patch set would fix some trivial issue too.
But it shouldn't be seen by anyone yet XD